jwt/__init__.py

```diff
@@ -38,9 +38,24 @@ class EncodeError(Exception): pass
     'RS256': lambda msg, key, sig: key.verify(hashlib.sha256(msg).digest(), sig, 'sha256'),
     'RS384': lambda msg, key, sig: key.verify(hashlib.sha384(msg).digest(), sig, 'sha384'),
     'RS512': lambda msg, key, sig: key.verify(hashlib.sha512(msg).digest(), sig, 'sha512'),
-    'none': lambda msg, key, sig: True
+    'none': lambda msg, key, sig: check_none_verify(msg, key, sig)
 }
 
+SUPPORTED_ALGOS = tuple(verifying_methods.keys())
+ALLOWED_ALGOS = sorted(SUPPORTED_ALGOS)
+# Remove the 'none' type from our particular usage
+del ALLOWED_ALGOS[len(ALLOWED_ALGOS) - 1] 
```

Comment:
I'm not concerned about spec compliance since this fork is years out of date and used by only a very small number of Mozilla projects(at least I sincerely hope that is the case).  If we are backporting changes from the main repo then we need to add support for our needs to the upstream and this repo needs to die.